### PR TITLE
Few improvements to config

### PR DESCRIPTION
**Pull request by Kurtchekov**
Merged Epic Fight's config capabilities with this mod's NBT tags for improved customisation. Now using JSON schema for describing config file, easier to add features later on. Default settings are now available in the assets folder and can be overwritten by kubejs, for example.

### CODE REVIEW

**Comment by Kurtchekov** [comment]
I've been playing with my spouse for several hours now and couldn't find any bugs, seems pretty stable so far. Config has some values that I came up with, not balanced yet.

File: src/main/java/github/pitbox46/epicfightnbt/Config.java [hunk 1]
```diff
@@ -35,88 +36,65 @@
...
-                            "  \"crossbow/stock\": {\n" +
-                            "    \"crossbow/basic_stock\": \"crossbow\"\n" +
-                            "  },\n" +
-                            "  \"bow/string\": {\n" +
-                            "    \"bow/basic_string\": \"bow\"\n" +
-                            "  }\n" +
-                            "}";
-                    writer.write(json);
-                    writer.close();
+                InputStreamReader defaults = new InputStreamReader(Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResourceAsStream("assets/epicfightnbt/epicfightnbt.json")));
```
Comment (pitbox46) [inline src/main/java/github/pitbox46/epicfightnbt/Config.java]: You got rid of the default config. The default config works like a regular Forge default server config where the config is applied to any new world that is created. Without this, players would have to copy paste their customized config for every new world. 

(I have not tested this code)
```suggestion
                InputStreamReader defaults = null;
                Path defaultConfigPath = FMLPaths.GAMEDIR.get().resolve(FMLConfig.defaultConfigPath()).resolve("epicfightnbt.json");
                if (!Files.exists(defaultConfigPath)) {
                    Files.copy(Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResourceAsStream("assets/epicfightnbt/epicfightnbt.json")), defaultConfigPath);
                }
                defaults = new InputStreamReader(Files.newInputStream(defaultConfigPath));
```

File: src/main/resources/assets/epicfightnbt/epicfightnbt.json [hunk 2]
```diff
@@ -0,0 +1,134 @@
+{
+  "weapons": [
```
Comment (pitbox46) [inline src/main/resources/assets/epicfightnbt/epicfightnbt.json]: Can't we simplify this to be in the form
```
{
    "double/head_left": {
        "subtypes": [
            {
                "armor_ignorance": 20,
                "hit_at_once": 0,
                "impact": 4,
                "weapon_type": "axe",
                "nbt_value": "double/basic_hammer_left"
            },
            ...
        ]
    },
    ...
}
```
In `NBTCategorySchema` would GSON be able to deserialize a map into this form automatically? If not that's fine, this is a minor thing.

**Review by pitbox46 — CHANGES_REQUESTED** [review]
This is a very good change, but there are some minor issues with it. It would also be nice to have a data fixer that converts from the old config system to the new one, but I don't think that's really worth the hassle.

File: src/main/java/github/pitbox46/epicfightnbt/Config.java (hunk 1 above)
Comment (Kurtchekov) [reply src/main/java/github/pitbox46/epicfightnbt/Config.java]: Oh, sorry, I didn't mean to, I'm still stuck to 1.12 and don't know all the new cool features of later versions. Re-implementing that ASAP.

File: src/main/resources/assets/epicfightnbt/epicfightnbt.json (hunk 2 above)
Comment (Kurtchekov) [reply src/main/resources/assets/epicfightnbt/epicfightnbt.json]: Working on it now

**Review by pitbox46 — APPROVED** [review]
Everything looks good now! Thanks for this PR!

**Pull request merged**